tests/extmod/machine_uart_irq_txidle.py

```diff
@@ -0,0 +1,69 @@
+# Test machine.UART.IRQ_TXIDLE firing after a transmission.
+# Does not require any external connections.
+
+try:
+    from machine import UART
+
+    UART.IRQ_TXIDLE
+except (ImportError, AttributeError):
+    print("SKIP")
+    raise SystemExit
+
+import time, sys
+
+# Configure pins based on the target.
+if "rp2" in sys.platform:
+    uart_id = 0
+    tx_pin = "GPIO0"
+    rx_pin = "GPIO1"
+    min_window = 1
+elif "samd" in sys.platform and "ItsyBitsy M0" in sys.implementation._machine:
+    uart_id = 0
+    tx_pin = "D1"
+    rx_pin = "D0"
+    # For SAMD delay_ms has to be used for the trailing window, and the
+    # mininmal time is 11 ms to allow for scheduling.
+    min_window = 11
+elif "samd" in sys.platform and "ItsyBitsy M4" in sys.implementation._machine:
+    uart_id = 3
+    tx_pin = "D1"
+    rx_pin = "D0"
+    min_window = 11
+elif "mimxrt" in sys.platform and "Teensy 4" in sys.implementation._machine:
+    uart_id = 1
+    tx_pin = None
+    min_window = 0
+elif "nrf" in sys.platform:
+    uart_id = 0
+    tx_pin = None
+    min_window = 0
+else:
+    print("SKIP")
+    raise SystemExit
+
+
+def irq(u):
+    print("IRQ_TXIDLE:", bool(u.irq().flags() & u.IRQ_TXIDLE))
```

Comment:
Can this be a `u.irq().flags() == u.IRQ_TXIDLE`?  Which ports require the `&`?

Reply:
I just made that identical to the other test scripts, which use the `bool(a & b)` term.

Reply:
It's stricter to test with `==`, which I would prefer to do if possible.

The tests that require `bool(a & b)` probably need it for a (good?) reason.

Reply:
Changed back for IRQ_TXIDLE.